docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,59 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow all
+kind of customisations since it would make the system even more complex to test
+and maintain over time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases and applications running on the top of it.
+
+It can be:
+
+- The BootstrapConfig,
+- The various roles and taints we set on the node objects of the cluster
+- The labels and annotations we would set on all K8s objects
+- The configurations associated to solutions, such as the list of available
+  solutions, the environments and namespaces created for a solution
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted
+and resistant to various type of events such as node reboot, upgrade, downgrade
+, backup, restore.
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- Number of replicas for the Prometheus, Alert Manager, Grafana or Dex
+- Other items will appear as we add new customisable features in MetalK8s
+
+Depending on the setting, impact of the customisation can be at cluster
+level or namespace level (not sure it is needed).
```

Comment:
For me there is no customisation for a namespace, it's more on the solution level if needed

Reply:
Agreed, let's remove this sentence.